package/MDAnalysis/core/groups.py

```diff
@@ -2258,7 +2258,7 @@ def residues(self):
         """A sorted :class:`ResidueGroup` of the unique
         :class:`Residues<Residue>` present in the :class:`AtomGroup`.
         """
-        rg = self.universe.residues[unique_int_1d(self.resindices)]
+        rg = self.universe.residues[unique_int_1d(np.array((self.resindices),dtype='int64'))]
```

Comment:
Use `ndarray.astype()` to save a conversion when it is already correctly int64, and using the dtype object directly:
```suggestion
        rg = self.universe.residues[unique_int_1d(self.resindices.astype(np.int64)]
```